### PR TITLE
[Model monitoring] Controller kafka stream fix

### PR DESCRIPTION
**Pull request by royischoss**
Fixes for controller stream using kafka part of requirement [ML- 5098](https://iguazio.atlassian.net/browse/ML-5098)

### CODE REVIEW

File: mlrun/model_monitoring/stream_processing.py [hunk 1]
```diff
@@ -270,11 +271,8 @@ def apply_parquet_target():
         apply_parquet_target()
 
         # controller branch
-        def apply_push_controller_stream():
-            stream_uri = get_stream_path(
-                project=self.project,
-                function_name=MonitoringFunctionNames.APPLICATION_CONTROLLER,
-            )
+        def apply_push_controller_stream(stream_uri: str):
+            logger.info("stream uri", stream_uri=stream_uri)
```
Comment (davesh0812) [inline mlrun/model_monitoring/stream_processing.py]: edit or delete this log

**Review by davesh0812 — APPROVED** [review]
LGTM

File: mlrun/model_monitoring/stream_processing.py [hunk 2]
```diff
@@ -285,17 +282,31 @@ def apply_push_controller_stream():
...
+                        "Brokers must be a list or str check controller stream uri"
+                    )
                 graph.add_step(
                     ">>",
                     "controller_stream_kafka",
-                    path=stream_uri,
+                    path=path,
                     kafka_brokers=brokers,
-                    sharding_func=ControllerEvent.ENDPOINT_ID,
+                    _sharding_func="kafka_sharding_func",
```
Comment (gtopper) [inline mlrun/model_monitoring/stream_processing.py]: Better add a TODO to revert this change once it's fixed in storey.

Comment (gtopper) [reply mlrun/model_monitoring/stream_processing.py]: Please resolve after the relevant change is pushed.

File: mlrun/model_monitoring/stream_processing.py [hunk 3]
```diff
@@ -285,17 +282,31 @@ def apply_push_controller_stream():
...
+                        "Brokers must be a list or str check controller stream uri"
+                    )
                 graph.add_step(
                     ">>",
                     "controller_stream_kafka",
-                    path=stream_uri,
+                    path=path,
                     kafka_brokers=brokers,
-                    sharding_func=ControllerEvent.ENDPOINT_ID,
+                    _sharding_func="kafka_sharding_func",  # TODO: remove this when storey handle str key
```
Comment (gtopper) [inline mlrun/model_monitoring/stream_processing.py:305]: https://github.com/mlrun/storey/pull/553
